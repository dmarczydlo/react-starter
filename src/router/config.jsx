import React from 'react';
import {Route} from 'react-router-dom';
import Home from '../sites/Home';
import Form from '../sites/Form';


export const routes = [
    {
        component: Home,
        exact: true,
        path: '/'
    },
    {
        component: Form,
        path: '/form'
    }
];

export const RouteWithSubRoutes = (route, key) => (
    <Route
        exact={route.exact}
        key={key}
        path={route.path}
        render={(props) => (
            <route.component
                {...props}
                routes={route.routes}
            />
        )}
    />
);
