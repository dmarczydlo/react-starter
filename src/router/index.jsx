import React from 'react';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import {RouteWithSubRoutes, routes} from './config';
import NotFound from '../sites/notFound';
import Layout from '../layout';

const RouterElements = () => {
    return (
        <Router>
            <Layout>
                <Switch>
                    {routes.map((route, key) => (
                        <RouteWithSubRoutes {...route} key={key} />
                    ))}

                    <Route component={NotFound} />
                </Switch>
            </Layout>
        </Router>
    );
};

export default RouterElements;
