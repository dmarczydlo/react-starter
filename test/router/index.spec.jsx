import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import Index from '../../src/router/index';
import {routes, RouteWithSubRoutes} from '../../src/router/config';

describe('Router index', () => {
    let wrapper;

    it('should be rendered', () => {
        wrapper = global.shallow(<Index />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Config should be rendered', () => {
        const route = routes[0];
        expect(RouteWithSubRoutes(route)).not.toBeNull();
        wrapper = global.shallow(<Router>{RouteWithSubRoutes(route)}</Router>);
        expect(wrapper).toMatchSnapshot();
    });

    it('Config should be rendered without layout', () => {
        const Test = () => <span>test</span>;
        const route = {
            path: '/test/',
            component: Test,
            layout: false
        };
        expect(RouteWithSubRoutes(route)).not.toBeNull();
        wrapper = global.shallow(<Router>{RouteWithSubRoutes(route)}</Router>);
        expect(wrapper).toMatchSnapshot();
    });

});
