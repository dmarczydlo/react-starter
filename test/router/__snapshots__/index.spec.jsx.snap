// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router index Config should be rendered 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
</Router>
`;

exports[`Router index Config should be rendered without layout 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    path="/test/"
    render={[Function]}
  />
</Router>
`;

exports[`Router index should be rendered 1`] = `
<BrowserRouter>
  <index>
    <Switch>
      <RouteWithSubRoutes
        component={[Function]}
        exact={true}
        key="0"
        path="/"
      />
      <RouteWithSubRoutes
        component={[Function]}
        key="1"
        path="/child/:id"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </index>
</BrowserRouter>
`;
